the_best_student_Xabibullo/xabibullo_sql.sql

```diff
@@ -0,0 +1,145 @@
+-- Kurslar jadvali
+CREATE TABLE IF NOT EXISTS courses (
+    course_id SERIAL PRIMARY KEY,
+    course_name VARCHAR(100),
+    credit_hours INTEGER
+);
+
+-- Talabalar ro'yxati jadvali
+CREATE TABLE IF NOT EXISTS enrollments (
+    enrollment_id SERIAL PRIMARY KEY,
+    student_id INTEGER REFERENCES student(student_id),
+    course_id INTEGER REFERENCES courses(course_id),
+    grade INTEGER
+);
+
+-- Kurslar uchun yangi ma'lumotlar
+INSERT INTO courses (course_name, credit_hours) 
+VALUES 
+    ('Algebra', 3), 
+    ('World History', 4), 
+    ('Astronomy', 3), 
+    ('Organic Chemistry', 4), 
+    ('Microbiology', 3), 
+    ('Creative Writing', 2), 
+    ('Data Structures', 3), 
+    ('Ethics', 2), 
+    ('Finance', 3), 
+    ('Sociology', 2);
+
+-- Ro'yxatdan o'tgan talabalar uchun yangi yozuvlar
+INSERT INTO enrollments (student_id, course_id, grade) 
+VALUES 
+    (1, 1, 4), (1, 2, 5), (1, 3, 3),
+    (2, 2, 4), (2, 4, 5), (2, 6, 3),
+    (3, 3, 3), (3, 5, 5), (3, 7, 4),
+    (4, 8, 4), (4, 1, 3), (4, 6, 5),
+    (5, 7, 3), (5, 9, 5), (5, 10, 4),
+    (6, 1, 4), (6, 3, 5), (6, 8, 3),
+    (7, 2, 3), (7, 4, 4), (7, 9, 5),
+    (8, 6, 4), (8, 8, 5), (8, 10, 3),
+    (9, 3, 5), (9, 7, 3), (9, 10, 4),
+    (10, 4, 4), (10, 5, 5), (10, 9, 3);
+
+-- Talabalar haqida ma'lumotni ko'rsatish
+SELECT first_name, last_name, birthdate 
+FROM student;
+
+-- Matematika kursiga yozilgan talabalar
+SELECT s.first_name, s.last_name 
+FROM student s
+JOIN enrollments e ON s.student_id = e.student_id
+JOIN courses c ON e.course_id = c.course_id
+WHERE c.course_name = 'Algebra';
+
+-- Bahosi 4 dan past bo'lgan talabalar
+SELECT s.first_name, s.last_name 
+FROM student s
+JOIN enrollments e ON s.student_id = e.student_id
+WHERE e.grade < 4;
+
+-- Talabalar va ularning kurslari
+SELECT s.first_name, s.last_name, c.course_name 
+FROM student s
+JOIN enrollments e ON s.student_id = e.student_id
+JOIN courses c ON e.course_id = c.course_id;
+
+-- Kurslarga yozilmagan talabalar
+SELECT s.first_name, s.last_name 
+FROM student s
+LEFT JOIN enrollments e ON s.student_id = e.student_id
+WHERE e.enrollment_id IS NULL;
+
+-- Har bir kursga yozilgan talabalar soni
+SELECT c.course_name, COUNT(e.student_id) AS student_count 
+FROM courses c
+LEFT JOIN enrollments e ON c.course_id = e.course_id
+GROUP BY c.course_name;
+
+-- Eng ko'p talaba yozilgan kurs
+SELECT c.course_name 
+FROM courses c
+LEFT JOIN enrollments e ON c.course_id = e.course_id
+GROUP BY c.course_name
+ORDER BY COUNT(e.student_id) DESC
+LIMIT 1;
+
+-- Talabalarni familiya bo'yicha tartiblash
+SELECT first_name, last_name 
+FROM students
+ORDER BY last_name;
+
+-- 2015-yildan keyin tug'ilgan va tarix kursiga yozilgan talabalar
+SELECT s.first_name, s.last_name 
+FROM students s
+JOIN enrollments e ON s.student_id = e.student_id
+JOIN courses c ON e.course_id = c.course_id
+WHERE c.course_name = 'World History' AND s.birthdate > '2015-01-01';
```

Comment:
birthdate and enrolled_year are not the same
you don't have enrolled_year in your students table
you have to
- Find all students enrolled after 2015 and enrolled in the History course.